src/Protocol/ChunkDataSerializer.cpp

```diff
@@ -196,12 +196,16 @@ inline void cChunkDataSerializer::Serialize47(const int a_ChunkX, const int a_Ch
 	m_Packet.WriteBEInt32(a_ChunkX);
 	m_Packet.WriteBEInt32(a_ChunkZ);
 	m_Packet.WriteBool(true);      // "Ground-up continuous", or rather, "biome data present" flag
-	m_Packet.WriteBEUInt16(Bitmask.first);
+
+	// Minecraft 1.8 does not like completely empty packets
+	// Send one completely empty chunk section if this is the case
+	m_Packet.WriteBEUInt16(Bitmask.first ? Bitmask.first : 1);
```

Comment:
Please explain the magic number. Does this "1" mean the chunk is empty? Or is this just a placeholder.

Reply:
The value means that the chunk has the lowest chunk section (16x16x16) non-empty, and the rest is empty. The following code is modified to transmit a single chunk section with all blocks being air.